pkg/registry/certificates/certificates/strategy.go

```diff
@@ -357,7 +357,7 @@ func (csrApprovalStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.
 		// Determine if takeover is enabled
 		takeover := utilfeature.DefaultFeatureGate.Enabled(features.DeclarativeValidationTakeover)
 
-		const validationIdentifier = "csr_approval_update"
+		const validationIdentifier = "certificatesigningrequest_approve_update"
```

Comment:
the subresource is approval so this should be: certificatesigningrequest_approval_update